dali/base/dadfs.cpp

```diff
@@ -12892,6 +12923,7 @@ IPropertyTreeIterator *deserializeFileAttrIterator(MemoryBuffer& mb, unsigned nu
             attr->setPropInt64(getDFUQResultFieldName(DFUQRFsize), attr->getPropInt64(getDFUQResultFieldName(DFUQRForigsize), -1));//Sort the files with empty size to front
             setRecordCount(attr);
             setIsCompressed(attr);
+            setCost(attr);
```

Comment:
function: I think there'll be a problem here if the file is on >1 plane/filegroup.

If it is, each copy is returned/listed individually, and duplicateFileAttrOnOtherNodeGroup is called to clone the deserialized properties (which will include cost).
It looks like as it stands, that will mean each file copy will appear to have the sum total cost, not the individual cost on the plane that the copy resides on.

I think the call here should say something like:
```
const char *firstNodeGroup = attr->queryProp(getDFUQResultFieldName(DFUQRFnodegroup));
if (!isEmptyString(firsNodeGroup))
    setCost(attr, firstNodeGroup);
```
( and setCost changed so it deals with a specific node group ).

Then in duplicateFileAttrOnOtherNodeGroup it can call setCost again for the node group it has moved on to.

Reply:
@jakesmith if the firstNodeGroup is empty, it isn't going to calculate any cost for the file. Is that ok?  When is this likely to be empty?

Reply:
I think that's correct. There are 2 ways it could be empty afaics:
1) for a superfile, where either it has 0 subs, or >1 (if >1, then it's the separate issue of handling superfiles in general).
2) the groups have been filtered out by nodeGroupFilter, during deserialization, i.e. resulting in fileNodeGroups container no groups.

Reply:
>Then in duplicateFileAttrOnOtherNodeGroup it can call setCost again for the node group it has moved on to.

setCost needs recalling as it walks other filtered groups.